packages/flutter/lib/src/widgets/container.dart

```diff
@@ -102,6 +110,10 @@ class Container extends StatelessWidget {
     assert(padding == null || padding.isNonNegative);
     assert(decoration == null || decoration.debugAssertIsValid());
     assert(constraints == null || constraints.debugAssertIsValid());
+    assert(color == null || decoration == null,
```

Comment:
Can't we just merge them? Then the only thing we need to document is that if decoration specifies color too, it takes precedence or something